pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -0,0 +1,345 @@
+		rc.watchCancel()
+		delete(c.clients, clusterName)
+	}
+}
+
+func (c *clustersReconciler) setRemoteClientConfig(ctx context.Context, clusterName string, kubeconfig []byte) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	client, found := c.clients[clusterName]
+	if !found {
+		client = newRemoteClient(c.client, c.wlUpdateCh)
+		if c.builderOverride != nil {
+			client.builderOverride = c.builderOverride
+		}
+		c.clients[clusterName] = client
+	}
+
+	if err := client.setConfig(c.rootContext, kubeconfig); err != nil {
+		ctrl.LoggerFrom(ctx).Error(err, "reading local workload")
+		delete(c.clients, clusterName)
+		return err
+	}
+	return nil
+}
+
+func (a *clustersReconciler) controllerFor(acName string) (*remoteClient, bool) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	c, f := a.clients[acName]
+	return c, f
+}
+
+func (c *clustersReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	cluster := &kueuealpha.MultiKueueCluster{}
+	log := ctrl.LoggerFrom(ctx)
+
+	err := c.client.Get(ctx, req.NamespacedName, cluster)
+	if client.IgnoreNotFound(err) != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err != nil || !cluster.DeletionTimestamp.IsZero() {
+		c.stopAndRemoveCluster(req.Name)
+		return reconcile.Result{}, nil
+	}
+
+	// get the kubeconfig
+	kubeConfig, retry, err := c.getKubeConfig(ctx, &cluster.Spec.KubeConfig)
+	if retry {
+		return reconcile.Result{}, err
+	}
+	if err != nil {
+		log.Error(err, "reading kubeconfig")
+		return reconcile.Result{}, c.updateStatus(ctx, cluster, false, "BadConfig", err.Error())
+	}
+
+	if err := c.setRemoteClientConfig(ctx, cluster.Name, kubeConfig); err != nil {
+		log.Error(err, "setting kubeconfig")
+		return reconcile.Result{}, c.updateStatus(ctx, cluster, false, "ClientConnectionFailed", err.Error())
+	}
+	return reconcile.Result{}, c.updateStatus(ctx, cluster, true, "Active", "Connected")
+}
+
+func (c *clustersReconciler) getKubeConfig(ctx context.Context, ref *kueuealpha.KubeConfig) ([]byte, bool, error) {
+	sec := corev1.Secret{}
+	secretObjKey := types.NamespacedName{
+		Namespace: c.configNamespace,
+		Name:      ref.Location,
+	}
+	err := c.client.Get(ctx, secretObjKey, &sec)
+	if err != nil {
+		return nil, !apierrors.IsNotFound(err), err
+	}
+
+	kconfigBytes, found := sec.Data[kueuealpha.MultiKueueConfigSecretKey]
+	if !found {
+		return nil, false, fmt.Errorf("key %q not found in secret %q", kueuealpha.MultiKueueConfigSecretKey, ref.Location)
+	}
+
+	return kconfigBytes, false, nil
+}
+
+func (c *clustersReconciler) updateStatus(ctx context.Context, cluster *kueuealpha.MultiKueueCluster, active bool, reason, message string) error {
+	newCondition := metav1.Condition{
+		Type:    kueuealpha.MultiKueueClusterActive,
+		Status:  metav1.ConditionFalse,
+		Reason:  reason,
+		Message: message,
+	}
+	if active {
+		newCondition.Status = metav1.ConditionTrue
+	}
+
+	// if the cluster is connected and remains that way, skip the status update
```

Comment:
Are you still going to call update if the Cluster is disconnected and remains that way?